redfish/provider/resource_redfish_bios.go

```diff
@@ -1,5 +1,472 @@
+				MarkdownDescription: "List of server BMCs and their respective user credentials",
+				Description:         "List of server BMCs and their respective user credentials",
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+					listvalidator.IsRequired(),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: RedfishServerSchema(),
+				},
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *BiosResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Trace(ctx, "resource_Bios create : Started")
+	// Get Plan Data
+	var plan models.Bios
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	service, err := NewConfig(r.p, &plan.RedfishServer)
+	if err != nil {
+		resp.Diagnostics.AddError("service error", err.Error())
+		return
+	}
+
+	state, diags := r.updateRedfishDellBiosAttributes(ctx, service, &plan)
+	if err != nil {
+		diags.AddError("Error running job %w", err.Error())
+	}
+	resp.Diagnostics.Append(diags...)
+
+	tflog.Trace(ctx, "resource_Bios create: updating state finished, saving ...")
+	// Save into State
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	tflog.Trace(ctx, "resource_Bios create: finish")
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *BiosResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	tflog.Trace(ctx, "resource_Bios read: started")
+	var state models.Bios
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	service, err := NewConfig(r.p, &state.RedfishServer)
+	if err != nil {
+		resp.Diagnostics.AddError("service error", err.Error())
+		return
+	}
+
+	err = r.readRedfishDellBiosAttributes(service, &state)
+	if err != nil {
+		diags.AddError("Error running job %w", err.Error())
+	}
+	resp.Diagnostics.Append(diags...)
+
+	tflog.Trace(ctx, "resource_Bios read: finished reading state")
+	// Save into State
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	tflog.Trace(ctx, "resource_Bios read: finished")
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *BiosResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Get state Data
+	tflog.Trace(ctx, "resource_Bios update: started")
+	var plan models.Bios
+
+	// Get plan Data
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	service, err := NewConfig(r.p, &plan.RedfishServer)
+	if err != nil {
+		resp.Diagnostics.AddError("service error", err.Error())
+		return
+	}
+
+	state, diags := r.updateRedfishDellBiosAttributes(ctx, service, &plan)
+	if err != nil {
+		diags.AddError("Error running job %v", err.Error())
```

Comment:
Same here . %v is not effective as a formatting directive here, it will show up to the user as-is
```suggestion
		diags.AddError("Error running job", err.Error())
```